### ISSUE TITLE
MagicNumber rule is too vague.

### ISSUE THREAD

**Issue by CodyEngel** [arturbosch/sonar-kotlin#49]
Consider a class with this definition: `class SomeClassWithDefault(val defaultValue: Int = 10)`

Currently Detekt will flag this saying it's a MagicNumber. However it doesn't seem to be in the spirit of Kotlin itself since the value is tied to a property it's not _actually_ a magic number. Are there any plans to update how this rule works so it doesn't flag default values?

**Comment by jolkdarr** [comment]
Hi

What do you think about the following code?
```kotlin
class Class1WithDefault(val defaultValue: Int = 10)
class Class1WithDefault(val defaultValue: Int = 10)
```

**Comment by arturbosch** [comment]
Hey, I linked this issue in the detekt slack channel to discuss this.
I think we haven't builtin an option to ignore default values because the name of the property does not tell enough why the value is 10. For example `class Person(val salary: Int = 5000)`: ok we know that 5000 must be the salary, but is it low or high or average? Writing `class Person(val salary: Int = SALARY_AT_JOB_START)` is more clear.

**Comment by jolkdarr** [comment]
If the value occurs only once in the unit, then one may assume it's not a magic number as suggested by @CodyEngel. Checking the rule would be just slightly more complicated.

**Comment by CodyEngel** [comment]
@jolkdarr / @arturbosch - within my own use case the values that were flagged was `initialHeight` and `initialWidth` for a data structure backed by a 2D array. I used the default value because it seemed like a more Kotlin way of doing things as opposed to having a companion object with `DEFAULT_HEIGHT` and `DEFAULT_WIDTH` just to add slightly more context.

As for the example of `class Person(val salary...` I would interpret that as "if we don't give this person a salary, it will default to 5000" which seems like more than enough documentation to me. If additional details are needed I don't see why it can't be covered in the class documentation.

**Comment by arturbosch** [comment]
I've created an issue at detekt to propose an ignore property for constructors. Will close this, your free to comment on the new issue :)
